src/io/flutter/run/daemon/FlutterAppService.java

```diff
@@ -45,17 +44,19 @@ private FlutterAppService(Project project) {
   /**
    * Start a Flutter app using the current Flutter SDK.
    * @param workDir The directory where the command should be run.
+   * @param programArgs Additional program args.
    * @param device The device to use.
    * @param mode The RunMode to use (release, debug, profile).
    * @param path Path to the Dart file containing the main method.
    */
   public FlutterApp startFlutterSdkApp(@NotNull String workDir,
+                                       @Nullable String programArgs,
                                        @Nullable FlutterDevice device,
                                        @NotNull RunMode mode,
                                        @Nullable String path)
     throws ExecutionException {
 
-    final GeneralCommandLine command = createFlutterSdkRunCommand(workDir, device, mode, path);
+    final GeneralCommandLine command = createFlutterSdkRunCommand(workDir, programArgs, device, mode, path);
```

Comment:
Suppose we move createFlutterSdkRunCommand() to SdkFields and just pass in the GeneralCommandLine as a parameter here? That we we build the arguments in one place.

(I was thinking about doing the same for Bazel, as well.)